packages/cli/src/lib/setup/setupUpgrade.ts

```diff
@@ -6,54 +6,73 @@
  *      MIT License
  *
  */
+import Debug from 'debug';
+import * as fs from 'fs-extra';
+import { tools, EXIT_CODES } from '@iobroker/js-controller-common';
+import semver from 'semver';
+import { Upload } from './setupUpload';
+import { Install } from './setupInstall';
+import rl from 'readline-sync';
+import tty from 'tty';
+import type { ObjectsInRedisClient } from '@iobroker/db-objects-redis/build/lib/objects/objectsInRedisClient';
+
+const debug = Debug('iobroker:cli');
+
+type IoPackDependencies = string[] | Record<string, any>[] | Record<string, any>;
+
+interface CLIUpgradeOptions {
+    processExit: (exitCode?: number) => void;
+    restartController: () => void;
+    getRepository: (repoName: string | undefined, params: Record<string, any>) => Record<string, any>;
+    objects: ObjectsInRedisClient;
+    params: Record<string, any>;
+}
 
-'use strict';
-const debug = require('debug')('iobroker:cli');
-
-/** @class */
-function Upgrade(options) {
-    const fs = require('fs-extra');
-    const { tools } = require('@iobroker/js-controller-common');
-
-    options = options || {};
-
-    if (!options.processExit) {
-        throw new Error('Invalid arguments: processExit is missing');
-    }
-    if (!options.restartController) {
-        throw new Error('Invalid arguments: restartController is missing');
-    }
-    if (!options.getRepository) {
-        throw new Error('Invalid arguments: getRepository is missing');
-    }
+export class Upgrade {
+    private readonly hostname = tools.getHostName();
+    private readonly upload: Upload;
+    private readonly install: Install;
+    private objects: ObjectsInRedisClient;
+    private readonly processExit: CLIUpgradeOptions['processExit'];
+    private readonly params: CLIUpgradeOptions['params'];
+    private readonly getRepository: CLIUpgradeOptions['getRepository'];
```

Comment:
I think it would be cleaner to extract those types into their own `type` definition and use them here and in `CLIUpgradeOptions`